### PR TITLE
[#48] Account activation

### PR DESCRIPTION
**Pull request by Monstarrrr**
#48

### CODE REVIEW

**Comment by Monstarrrr** [comment]
This breaks the temporary login page implementation I had made so this will have to be addressed otherwise it'll break & prevent deployments

File: frontend/app/(routes)/(auth)/register/page.tsx [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
     {
-      defaultValue: 'test',
       id: 'username',
       placeholder: 'Username',
+      value: '',
     },
     {
-      defaultValue: 'test@test.com',
       id: 'email',
       placeholder: 'Email',
```
Comment (seporterfield) [inline frontend/app/(routes)/(auth)/register/page.tsx:18]: the new placeholder values are a lot better!

File: frontend/app/_components/form.tsx [hunk 2]
```diff
@@ -1,20 +1,70 @@
...
+    inputsErrors,
+    buttonLabel,
+    successMessage,
+  } = props
+  // Renaming to avoid confusion with fields ids
+  const formId = id
   const [inputsState, setInputsState] = useState(inputsFields)
   const [globalFormErrors, setGlobalFormErrors] = useState<string[] | null>(null)
 
+  // Load cached inputs if they exist
```
Comment (seporterfield) [inline frontend/app/_components/form.tsx:27]: This is cool

**Review by seporterfield — APPROVED** [review]

File: frontend/app/_types/inputs.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 export type TextInputType = {
   id: string
   placeholder: string
-  defaultValue?: string
+  value: string | number
```
Comment (Monstarrrr) [inline frontend/app/_types/inputs.ts:4]: Just realized this InputType is Text specifically (hence the type name), and I won't make it a general Input type that accepts text & numbers & date etc... so I need to remove the "number" here and some of the type's types (date, number...)

**Pull request merged**